packages/web/components/Modal/CurateSpecialtiesDeleteModal/index.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useRouter } from 'next/router';
+import { Container, Question, ButtonsContainer, Button } from './styles';
+import { updateCategoriesReviewer } from '../../../services';
+import { toast } from '../../Toast';
+
+const CurateSpecialtiesDeleteModal = ({ categories, speciality, closeModal }) => {
+	const router = useRouter();
+
+	const onRemove = async () => {
+		const specialties = categories.filter((c) => !speciality.includes(c.id)).map((c) => c.id);
+		const reviewer = await updateCategoriesReviewer({
+			categories: specialties,
+		});
+
+		if (reviewer) {
+			closeModal();
+			router.reload();
+			toast.success('Solicitação enviada com sucesso');
+		} else {
+			toast.error('Ocorreu um erro, recarregue a página e tente novamente');
+		}
```

Comment:
```suggestion
		const specialties = categories.filter((category) => !speciality.includes(category.id)).map((category) => category.id);
		const reviewer = await updateCategoriesReviewer({
			categories: specialties,
		});

		if (reviewer) {
			closeModal();
			router.reload();
			return toast.success('Solicitação enviada com sucesso');
		}
		
		return toast.error('Ocorreu um erro, recarregue a página e tente novamente');
```

Aqui para evitar if / else, é preferível return early https://medium.com/swlh/return-early-pattern-3d18a41bba8

Reply:
Feito!